controllers/front/return.php

```diff
@@ -65,7 +65,19 @@ public function postProcess()
         }
 
         try {
-            $this->assertTransaction($orderId);
+            /** @var SaferPayTransactionAssertion $transactionAssert */
+            $transactionAssert = $this->module->getService(SaferPayTransactionAssertion::class);
+            $assertionResponse = $transactionAssert->assert($cartId);
+
+            if (!$orderId) {
+                $paymentMethod = $assertionResponse->getPaymentMeans()->getBrand()->getPaymentMethod();
+
+                /** @var CheckoutProcessor $checkoutProcessor **/
+                $checkoutProcessor = $this->module->getService(CheckoutProcessor::class);
+                $checkoutProcessor->processCreateOrderAfterAuthorization(new \Cart($cartId), $paymentMethod);
```

Comment:
could just pass cartID

Reply:
we need to create an order from cart here